test/unit/ngsiv2/amqpBinding-test.js

```diff
@@ -0,0 +1,320 @@
+                .reply(204);
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/secondSingleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a two update context requests to the Context Broker one with each attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs', new Buffer('temperature|23#humidity|98'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+    describe('When multiple groups of measures arrive, with multiple attributes, to a Device routing key', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/multipleMeasure.json'))
+                .reply(204);
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/secondMultipleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a two update context requests to the Context Broker one with each attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs',
+                new Buffer('temperature|23|humidity|98#temperature|16|humidity|34'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a measure with a timestamp arrives with an alias to TimeInstant', function() {
+        var provisionProduction = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/deviceProvisioning/provisionTimeInstant.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        beforeEach(function(done) {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext')
+                .reply(200, '{}')
+                .post('/v2/entities/TimeInstant%20Device/attrs')
```

Comment:
No payload is checked in the mock for the `POST /v2/entities/TimeInstant%20Device/attrs` request?

I'm fact, I'm not sure whta this test case is testing, I don't see the usual `contextBrokerMock.done()`. Could you elaborate, please?

Reply:
Fixed

` utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantDuplicated.json'))
`

Reply:
Fixed

` utils.readExampleFile('./test/unit/ngsiv2/contextRequests/timeInstantDuplicated.json'))
`